application/modules/sessions/controllers/Sessions.php

```diff
@@ -187,7 +187,11 @@ public function passwordreset($token = null)
                 $email_message = $this->load->view('emails/passwordreset', array(
                     'resetlink' => $email_resetlink
                 ), true);
-                $email_from = 'system@' . preg_replace("/^[\w]{2,6}:\/\/([\w\d\.\-]+).*$/", "$1", base_url());
+
+                $email_from = get_setting('smtp_mail_from');
+                if (emtpy($email_from)) {
```

Comment:
@kalimerre this is why we check if the database is empty. If it is, it reverts to the old behaviour of using the ``system@`` for backward compatiblity. So that if a user updates, nothing will change in the behaviour.

Reply:
I saw that and it’s fine for me. The only thing which is not fine is that in the form in the settings the first time it will appears as empty/blank because there is no value in the DB. So user may think it will be empty or something else whereas it should shown system@ as default while going to the form for the first time. 

I don’t know if I’m very clear I’m on my phone so it’s hard to explain ;).

Reply:
Thanks for the hint. The settings label needs to be changed to something more clear, like "From address for system emails" or something similar.
I would not explicitly use "for password reset" emails to continue using the setting for other internal emails like for notifications.

Reply:
@kalimerre I understand. So maybe something like using ``placeholder=`` would be sensible, so that the user knows, that ``system@`` will be used as default, if he enters nothing.

@Kovah I happily can change this.

Reply:
Fine for me if it’s good for @Kovah too ;)